Advanced Pandas slides

cunningjames opened the pull request:
Closes #20.

Most of this is either new or substantially revised to fit the 45 minute time limit. It's possible some of this code might be confusing ... happy to revise if you think it necessary.

cunningjames committed:
initial commit

augustopher commented on notebooks/11-Advanced-Pandas.ipynb:
```diff
@@ -0,0 +1,2864 @@
+    "\n",
+    "> ...it’s very hard to predict whether [an operation] will return a view or a copy.\n",
+    "\n",
+    "I recommend reading the following section:\n",
+    "https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
```
I would make this a hyperlink to be cleaner, but not a big deal.

augustopher commented on notebooks/11-Advanced-Pandas.ipynb:
```diff
@@ -0,0 +1,2864 @@
+    "tags": []
+   },
+   "source": [
+    "I want you to get a sense for how this is possible, but if you don't follow all of the Python code here that's fine. Just be aware of the capability so you can research it further."
```
Maybe add **`Note:`** here, to make sure this call-out is very obvious.

augustopher commented on notebooks/11-Advanced-Pandas.ipynb:
```diff
@@ -0,0 +1,2864 @@
+    "tags": []
+   },
+   "source": [
+    "Now we need to split the data frame up into 4 portions, one ofr each core. For this we'll use the `array_split()` function from NumPy, which does exactly what it says, in this case returning a list of 4 data frames:"
```
```suggestion
    "Now we need to split the data frame up into 4 portions, one for each core. For this we'll use the `array_split()` function from NumPy, which does exactly what it says, in this case returning a list of 4 data frames:"
```

augustopher commented on notebooks/11-Advanced-Pandas.ipynb:
```diff
@@ -0,0 +1,2864 @@
+    "tags": []
+   },
+   "source": [
+    "## Joining Data Frames"
```
Should we move the joining section earlier? I would think we want to move from more common to less common use cases (with some flexibility for naturally building concepts), and joining data frames should be more relevant/common than parallelization.

augustopher commented on notebooks/11-Advanced-Pandas.ipynb:
```diff
@@ -0,0 +1,2864 @@
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "6500446b-2240-4d51-b875-604b7a5021d7",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "fragment"
+    },
+    "tags": []
+   },
+   "source": [
+    "We have several different options:"
+   ]
+  },
```
I feel like this fragment is a little unnecessary, at least hanging at the bottom of this slide.

augustopher commented on notebooks/11-Advanced-Pandas.ipynb:
```diff
@@ -0,0 +1,2864 @@
+    "tags": []
+   },
+   "source": [
+    "#### Left Outer Join"
```
I would up this to `h3`.

```suggestion
    "### Left Outer Join"
```

augustopher commented on notebooks/11-Advanced-Pandas.ipynb:
```diff
@@ -0,0 +1,2864 @@
+    "tags": []
+   },
+   "source": [
+    "#### Right Outer Join"
```
I would up this to `h3`.

```suggestion
    "### Right Outer Join"
```

augustopher commented on notebooks/11-Advanced-Pandas.ipynb:
```diff
@@ -0,0 +1,2864 @@
+    "tags": []
+   },
+   "source": [
+    "#### Full Outer Join"
```
I would up this to `h3`.

```suggestion
    "### Full Outer Join"
```

augustopher reviewed (changes requested):
Just some small typos and style suggestions, with one slightly larger re-org idea.

cunningjames committed:
addressing comments

cunningjames commented:
@augustopher, I think I've addressed all your comments.

I also added a section on cross joins. It's a bit more niche but it was the only join type I was leaving out, so I figured I may as well be complete.

cunningjames committed:
moved pandas import
